src/submission/validation-clinical/surgery.ts

```diff
@@ -199,3 +188,40 @@ function checkSurgeryTypeEquality(
 
 	return existingSurgeryType === therapyRecordSurgeryType;
 }
+
+// Prevents submitting duplicate records; allows updating existing records with new values
+function checkSurgeryDuplicateOrUpdate(
+	therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+	prevSurgeries: DeepReadonly<Therapy>[],
+	existingSurgery: DeepReadonly<Therapy> | undefined,
+	errors: SubmissionValidationError[],
+) {
+	// Clone Submitted Record, minus index key, which is not stored on Clinical Records
+	const submissionClone = (({ index, ...rest }) => ({ ...rest }))(therapyRecord);
+
+	// Determine if Submission is duplicating existing record, or if Submission is an update
+	// Sort insures [A,B,C] is not compared with [A,C,B]
+	const submissionValues = Object.values(submissionClone).sort();
+	const existingRecordValues = existingSurgery
+		? Object.values(existingSurgery.clinicalInfo).sort()
+		: [];
+	const submissionIsDuplicate =
+		submissionValues.length === existingRecordValues.length &&
+		isEqual(submissionValues, existingRecordValues);
+
+	if (submissionIsDuplicate || prevSurgeries.length > 1) {
+		// If there is duplicate surgery with the current submitter_specimen_id submitted before,
+		// or if there are more than one surgeries with the same submitter_specimen_id
+		// in the current submission, then invalid.
+		errors.push(
+			utils.buildSubmissionError(
+				therapyRecord,
+				DataValidationErrors.DUPLICATE_SUBMITTER_SPECIMEN_ID_IN_SURGERY,
+				DonorFieldsEnum.submitter_donor_id,
+				{
+					submitter_specimen_id: therapyRecord[SurgeryFieldsEnum.submitter_specimen_id],
+				},
+			),
+		);
+	}
```

Comment:
Related to previous comment... this is the only interaction with the `errors` array, can we instead have this function return the new items instead of mutating a variable that is passed in?

Reply:
Updated here:
https://github.com/icgc-argo/argo-clinical/pull/1162/commits/9a31dad1404f4c4dcc5e296063d71b825e930539

Could also use a `let errors; errors = [...errors, errorFunction()]` approach using reassignment
Half dozen of one, six of the other 🤷